### ISSUE TITLE
`form` panics when parsing empty modules

### ISSUE THREAD

**Issue by cr1901** [djmcgill/form#13]
## Problem
Consider the following two source files in the same directory, `lib.rs` and `foo.rs`:

`lib.rs`:

```rust
use foo::*;
pub mod foo;
```

`foo.rs`:

```rust
struct Bar;
```

When trying to split them, `form` panics because it [attempts to unwrap](https://github.com/djmcgill/form/blob/master/src/util.rs#L104) a `None` value:

```
William@DESKTOP-H0PMN4M MINGW64 ~/Projects/MCVE/form-mod
$ form -i lib.rs -o src/
[2021-04-14T16:45:09Z INFO  form::util] Started parsing the input as Rust. This can take a minute or two.
[2021-04-14T16:45:09Z INFO  form::util] Finished parsing
[2021-04-14T16:45:09Z INFO  form::util] Prepared target directory src/
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /rustc/760ce94c69ca510d44087291c311296f6d9ccdf5\src\libcore\macros\mod.rs:10:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

## Use Case
The actual use case I have for this is a patch I want to submit to `svd2rust` to support its `-g` option a bit better. The `-g` option emits a separate file called `generic.rs` to split the generic code from the peripherals.

Right now the `-g` option requires a user to manually add something like the following lines to `lib.rs` (preferably _after_ `form` has been run :)):

```rust
use generic::*;
pub mod generic;
```

Otherwise compilation will fail with errors like the following:

```sh
     error[E0433]: failed to resolve: maybe a missing crate `FieldReader`?
   --> src\port_1_2\p2ifg.rs:111:25
    |
111 |         P2IFG2_R(crate::FieldReader::new(bits))
    |                         ^^^^^^^^^^^ maybe a missing crate `FieldReader`?
```

My patch would automate adding the above code, but since my patch causes `form` to panic, I'd rather wait to submit my patch. I could fix this myself (ignore `mod` declarations without a body), but I wanted to open an issue in case you wanted to override/had a better idea :).

**Comment by djmcgill** [comment]
Yeah having `extract_mod` just return `None` when the `Item::Mod.content` is `None` sounds very reasonable and I'd welcome a PR since you're better positioned to test it. (Or I could do the ~2 line code change if you test it).

Thanks for the issue and I'm glad you're finding Form useful.

e: I think that there could potentially in future be issues with running Form on projects that already have e.g. half of a module structure and existing directory trees or whatever but am happy to cross that bridge when we get to it.

**Comment by cr1901** [comment]
@dirbaio on IRC mentioned a potential issue w/ my fix (I linked the issue to there):

```
(1:13:51 PM) dirbaio[m]: if `src/lib.rs` has `mod foo { mod bar { mod baz; }}`, [rustc] look for `src/baz.rs`
(1:13:51 PM) dirbaio[m]: form would move that to `src/foo/bar.rs`, so it'd look for `src/foo/baz.rs`
(1:14:12 PM) dirbaio[m]: probably not a problem for the svd2rust case though
...
(1:14:45 PM) dirbaio[m]: ignoring `mod foo;` is not 100% correct... but better than crashing at least :D
...
(1:16:00 PM) dirbaio[m]: form moving the `mod foo;` to another dir could make code no longer compile
```

So returning `None` and ignoring would indeed solve my use case, but there are other use cases that would not work properly. Is there a way to check the current `mod` nesting level and `panic` if it's nonzero (when an empty `mod` is found, of course :D) for now?

**Comment by djmcgill** [comment]
Okay I had a play around and while I _could_ start copying files around when we encounter nested modules I think for now just supporting top level itemless modules and going back to panicking on nested ones is fine for now. I took the time to the barest pass of code cleanup while I was at it too. Look good to you? https://github.com/djmcgill/form/pull/14

**Comment by cr1901** [comment]
@djmcgill Works fine for me! Thanks for the quick response time :D!

**Comment by djmcgill** [comment]
This is published as 0.8.0 now
